lib/backend/etcdv3/inline_cert_key.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code has been borrowed from etcd repository and has been modified
+// to provide support for inline certificates and keys for calicoctl.
+// Below are the github links for the files from which the code has been borrowed.
+
+// Copyright 2015 The etcd Authors
+//     https://github.com/etcd-io/etcd/blob/release-3.3/pkg/transport/listener.go
+
+// Copyright 2016 The etcd Authors
+//     https://github.com/etcd-io/etcd/blob/release-3.3/pkg/tlsutil/tlsutil.go
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdv3
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+// This struct is used to store the inline cert, key and CA cert data
+type TlsInlineCertKey struct {
+	Cert   string
+	Key    string
+	CACert string
+}
+
+// clientConfig() returns a config file with certificate data for client creation
```

Comment:
godoc doesn't match the method, please update method name and fix up the comment.  I think this is returning a TLS configuration based on the inline certs

Reply:
Acknowledged and updated comment.